game/game_application.cpp

```diff
@@ -47,7 +69,11 @@ GameApplication::GameApplication()
 	}
 	else
 	{
-		SceneLoader::GetSingleton()->loadScene(m_ApplicationSettings->getJSON()["startScene"], {});
+		std::string sceneName = m_ApplicationSettings->getJSON()["startScene"];
```

Comment:
Replace `std::string` with `String` everywhere you have used
https://github.com/sdslabs/Rootex/blob/e4cc841fb5bfb7ca848e671b005f3e92e3b1391d/rootex/common/types.h#L71

Reply:
done